src/main/java/org/zaproxy/zap/extension/jwt/JWTActiveScanRule.java

```diff
@@ -71,6 +79,29 @@ public void init() {
         }
     }
 
+    protected void scan(List<NameValuePair> nameValuePairs) {
+        if (nameValuePairs.isEmpty()
+                || nameValuePairs.get(0).getType() != NameValuePair.TYPE_HEADER) {
+            super.scan(nameValuePairs);
+        } else {
+            nameValuePairs = new ArrayList<>(nameValuePairs);
+            List<HttpHeaderField> headerFields = getBaseMsg().getRequestHeader().getHeaders();
+            for (HttpHeaderField headerField : headerFields) {
+                if (AUTHORIZATION_HEADER_KEY.equals(
+                        headerField.getName().toLowerCase(Locale.ROOT))) {
```

Comment:
`HttpRequestHeader.AUTHORIZATION.equalsIgnoreCase(headerField.getName())` which avoids creating new strings for the equals check.